front/components/assistant/details/tabs/AgentInfoTab/index.tsx

```diff
@@ -25,45 +47,61 @@ export function AgentInfoTab({
 
       {agentConfiguration.description && (
         <div className="text-sm text-foreground dark:text-foreground-night">
-          {agentConfiguration.description}
+          <Markdown
+            content={agentConfiguration.description}
+            forcedTextSize="text-sm"
+          />
         </div>
       )}
 
-      {agentConfiguration.scope !== "global" && (
+      {displayKnowledge && (
         <>
           <Page.Separator />
           <AssistantKnowledgeSection
             agentConfiguration={agentConfiguration}
             owner={owner}
           />
-
-          {agentConfiguration?.instructions ? (
-            <div className="dd-privacy-mask flex flex-col gap-4">
-              <div className="heading-lg text-foreground dark:text-foreground-night">
-                Instructions
-              </div>
-              <div
-                className={cn(
-                  "max-h-[400px] overflow-y-auto rounded-lg border border-border bg-muted-background px-3 py-2 " +
-                    "dark:border-border-night dark:bg-muted-background-night"
-                )}
-              >
-                <AgentMessageMarkdown
-                  content={agentConfiguration.instructions}
-                  owner={owner}
-                />
-              </div>
-            </div>
-          ) : (
-            <div className="text-sm text-muted-foreground dark:text-muted-foreground-night">
-              Instructions
-            </div>
-          )}
         </>
       )}
+
+      {displayInstructions && isString(agentConfiguration.instructions) && (
```

Comment:
super nitpick but we already check `isString(agentConfiguration.instructions)` inside displayInstructions?